rasa_nlu/cross_eval.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from rasa_nlu.converters import load_data
+from rasa_nlu.config import RasaNLUConfig
+from rasa_nlu.model import Trainer, TrainingData, Interpreter
+
+from sklearn.model_selection import StratifiedKFold
+from sklearn import metrics
+import numpy as np
+import logging
+import argparse
+import os
+from itertools import groupby
+from collections import defaultdict
+
+logger = logging.getLogger(__name__)
+
+def create_argparser():
+    parser = argparse.ArgumentParser(
+            description='cross evaluate a Rasa NLU pipeline')
+
+    parser.add_argument('-d', '--data', default=None,
+                        help="file containing training data")
+    parser.add_argument('-c', '--config', required=True,
+                        help="config file")
+    return parser
+
+
+def prepare_data(data, cutoff = 5):
+    """remove intent groups with less than cutoff instances."""
+    data = data.sorted_intent_examples()
+    logger.info("Raw data intent examples: {}".format(len(data)))
+
+    # count intents
+    list_intents = []
+    n_intents = []
+    for intent, group in groupby(data, lambda e: e.get("intent")):
+        size = len(list(group))
+        n_intents.append(size)
+        list_intents.append(intent)
+
+    # remove small intents groups
+    ind_delete = []
+    for ind in range(len(data)):
+        if data[ind].get("intent") in [list_intents[i] for i in range(len(list_intents)) if n_intents[i] < cutoff]:
```

Comment:
same here `enumerate(...)` instead of `range(len(...))` . please also move the list comprehension before the if statement, assign it to a var and then use it in the if